tensornetwork/backends/jax/jax_backend.py

```diff
@@ -49,37 +60,122 @@ def slice(self,
     if len(start_indices) != len(slice_sizes):
       raise ValueError("Lengths of start_indices and slice_sizes must be"
                        "identical.")
-    return self.jax.lax.dynamic_slice(tensor, start_indices, slice_sizes)
+    return libjax.lax.dynamic_slice(tensor, start_indices, slice_sizes)
+
+  def svd_decomposition(self,
+                        tensor: Tensor,
+                        split_axis: int,
+                        max_singular_values: Optional[int] = None,
+                        max_truncation_error: Optional[float] = None,
+                        relative: Optional[bool] = False
+                       ) -> Tuple[Tensor, Tensor, Tensor, Tensor]:
+    return decompositions.svd_decomposition(
+        jnp,
+        tensor,
+        split_axis,
+        max_singular_values,
+        max_truncation_error,
+        relative=relative)
+
+  def qr_decomposition(
+      self,
+      tensor: Tensor,
+      split_axis: int,
+  ) -> Tuple[Tensor, Tensor]:
+    return decompositions.qr_decomposition(jnp, tensor, split_axis)
+
+  def rq_decomposition(
+      self,
+      tensor: Tensor,
+      split_axis: int,
+  ) -> Tuple[Tensor, Tensor]:
+    return decompositions.rq_decomposition(jnp, tensor, split_axis)
+
+  def shape_tensor(self, tensor: Tensor) -> Tensor:
+    return tensor.shape
+
+  def shape_tuple(self, tensor: Tensor) -> Tuple[Optional[int], ...]:
+    return tensor.shape
+
+  def sparse_shape(self, tensor: Tensor) -> Tuple[Optional[int], ...]:
+    return self.shape_tuple(tensor)
+
+  def shape_prod(self, values: Tensor) -> Tensor:
+    return np.prod(values)
```

Comment:
is there a reason why this uses numpy instead of jax.numpy?

Reply:
Literally the only reason this calls numpy at all is because python doesn't have its own `prod` method.